src/content/en/fundamentals/primers/service-workers/a-basic-service-worker.markdown

```diff
@@ -0,0 +1,170 @@
+service worker's scope, and it's the lowest level of your website where the
+service worker can be used. Let's look at the registration code again, but with
+some changes to the `register()` parameters.
+
+{% highlight javascript %}
+// Does the browser support service workers?
+if ('serviceWorker' in navigator) {
+  // Yes
+  navigator.serviceWorker.register('/service-worker.js', {scope: '/'})
+    .then(function() {
+    // Actions not covered by this primer.
+  });
+} else {
+  // No, but this client should work anyway.
+}
+{% endhighlight %}
+
+This code is functionally identical to the example given in the last section. 
+Both examples specify that the scope of the service worker is the entire domain. 
+If you want to restrict the service worker to part of a site, simply specify it 
+in the scope. 
+
+For example, say that you have an auction site with different features for
+buyers  and sellers. The buyers section is at `example.com/buyers` and the
+sellers section  is at `example.com/sellers`. A service worker with a scope of
+`/buyers` it can only  serve clients under `example.com/buyers`. It cannot serve
+clients under  `example.com/sellers`. Similarly, a service worker with a scope
+of `/sellers` can  only serve clients under `example.com/sellers`.
+
+{% include shared/note.liquid list=page.notes.https-only %}
+
+## A service worker installs and activates
+
+Let’s look at what happens during the installation of the service worker. To set 
+up a service worker, you need to implement two events:
+
+* `install`
+* `activate`
+
+These events run one time for a service worker in a particular scope, no matter 
+how many clients use it.
+
+### Install
+
+Use the install event to get everything this service worker needs to operate.
+This could include initializing caches,  prefetching data, handling old versions
+of the current service worker, or  handling other service workers in
+the same scope.
+
+The install event is kicked off immediately after the script containing it is
+downloaded.
+
+{% highlight javascript %}
+self.addEventListener('install', function(installEvent) {
+  installEvent.waitUntil(
+    // Prefetch resources and initialize data stores.
+  );
+});
+{% endhighlight %}
+
+### Activate
+
+Use activate to clean up after a previous version of a service worker. This may
+include migrating data in ways that can't be done while the old service worker
+was active. This event is **not** called when a terminated service worker is
+revived.
+
+{% highlight javascript %}
+self.addEventListener('activate', function(activateEvent) {
+  activateEvent.waitUntil(
+    // Actions not covered by this primer.
+  );
+});
+{% endhighlight %}
+
+## The user navigates
+
+Now that the service worker is installed, it's doing something for our page,
+right? No quite. For the service worker to start handling requests, the user
+needs to navigate. Either you can wait for the user to open another page in the
```

Comment:
Or the service worker could call `clients.claim()` as part of the `activate` handler.

I can understand not wanting to get into the details of `clients.claim()` in the primer, but at the same time, the current language describes a default behavior that could be overridden as if it were the only behavior.